pkg/sql/sem/eval/context.go

```diff
@@ -927,6 +927,9 @@ type ReplicationStreamManager interface {
 		ctx context.Context,
 		req streampb.ReplicationProducerRequest,
 	) (streampb.ReplicationProducerSpec, error)
+
+	AuthorizeViaJob(ctx context.Context, streamID streampb.StreamID) error
```

Comment:
I'm wondering if `AuthorizeViaJob` and `AuthorizeViaReplicationPriv` need to be added to `ReplicationStreamManager`'s public interface. I see these methods are called by many of the crdb built ins, but most of the built ins are running authorization logic twice: once when the built in is called and once inside the replication stream manager's method implementation. Ideally we would authorize once. That helps clarify which layer of the implementation is responsible for authorization.

Reply:
> but most of the built ins are running authorization logic twice: once when the built in is called and once inside the replication stream manager's method implementation.
Maybe I'm misunderstanding you comment, but my intention was for `AuthorizeViaJob` and `AuthorizeViaReplicationPriv` to be called in the crdb built in, never within the stream ingestion manager's method implementation. Within the implementation, we merely check that that the function has been `Authorized` (perhaps this func should be private). Does this model make sense to you?

Reply:
Ahh. Yes. That makes sense. So the authorization in the built ins is load bearing, then the secondary check is there as an assertion.